guides/common/modules/proc_configuring-smart-proxy-server-with-custom-ssl-certificates-to-generate-and-sign-puppet-certificates.adoc

```diff
@@ -56,17 +56,31 @@ Retain a copy of the example `{foreman-installer}` command from the output for i
 --puppet-server true \
 --puppet-server-ca "true"
 ----
+. On {SmartProxyServer}, stop the Puppet server:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# puppetserver stop
```

Comment:
Real need - when I ran the `puppetserver` command, it has told me to stop the service first. And stopping it with another `puppetserver` command worked just fine.

Reply:
However, I was running the command on the Puppet CA server. As you say in another comment, that's not the right place for it. Is it safe to assume that it would ask me to stop the service on Puppet non-CA also?

Reply:
I also was confused by this, but then I've read the docs:
```
$ puppetserver ca generate --help
  If the `--ca-client` flag is passed, the cert will be generated
  offline, without using Puppet Server's signing code, and will add
  a special extension authorizing it to talk to the CA API. This can
  be used for regenerating the server's host cert, or for manually
  setting up other nodes to be CA clients. Do not distribute certs
  generated this way to any node that you do not intend to have
  administrative access to the CA (e.g. the ability to sign a cert).

  Since the `--ca-client` causes a cert to be generated offline, it
  should ONLY be used when Puppet Server is NOT running, to avoid
  conflicting with the actions of the CA service. This will be
  mandatory in a future release.
```